src/Symfony/Component/Validator/Constraints/CssColorValidator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+class CssColorValidator extends ConstraintValidator
+{
+    private const PATTERN_HTML5 = '/^#[0-9a-f]{6}$/i';
+    private const PATTERN_LONG = '/^#[0-9a-f]{8}$/i';
+    private const PATTERN_SHORT = '/^#[0-9a-f]{3,4}$/i';
+    private const PATTERN_NAMED_COLORS = '/^(black|red|green|yellow|blue|magenta|cyan|white)/i';
+
+    private const COLOR_PATTERNS = [
+        CssColor::VALIDATION_MODE_HTML5 => self::PATTERN_HTML5,
+        CssColor::VALIDATION_MODE_LONG => self::PATTERN_LONG,
+        CssColor::VALIDATION_MODE_SHORT => self::PATTERN_SHORT,
+        CssColor::VALIDATION_MODE_NAMED_COLORS => self::PATTERN_NAMED_COLORS,
+    ];
+
+    private $defaultMode;
+
+    public function __construct(string $defaultMode = CssColor::VALIDATION_MODE_HTML5)
```

Comment:
How is it possible to allow several patterns? (ie. "short or long" or "with or without alpha")

If we want a single pattern in mode, we should relax some patterns.

```
PATTERN_SHORT = '/^#[0-9a-f]{3,4}$/i';
PATTERN_LONG = '/^#([0-9a-f]{3,4}){2}$/i';
PATTERN_HEXA = '/^#([0-9a-f]{3,4}){1,2}$/i';

PATTERN_SHORT_ALPHA = '/^#[0-9a-f]{4}$/i';
PATTERN_LONG_ALPHA = '/^#[0-9a-f]{8}$/i';
PATTERN_HEXA_ALPHA = '/^#([0-9a-f]{4}){1,2}$/i';

PATTERN_SHORT_NO_ALPHA = '/^#[0-9a-f]{3}$/i';
PATTERN_LONG_NO_ALPHA = '/^#[0-9a-f]{6}$/i';
PATTERN_HEXA_NO_ALPHA = '/^#([0-9a-f]{3}){1,2}$/i';
```

Reply:
We could take example from CardScheme, and accept an array of modes (which could be renamed to `patterns`?), WDTY?